cpp/src/io/parquet/compact_protocol_reader.cpp

```diff
@@ -61,16 +63,15 @@ class parquet_field_list : public parquet_field {
   }
 
  public:
-  inline bool operator()(CompactProtocolReader* cpr, int field_type)
+  inline void operator()(CompactProtocolReader* cpr, int field_type)
   {
-    if (field_type != ST_FLD_LIST) { return true; }
+    CUDF_EXPECTS(field_type == ST_FLD_LIST, "expected list field");
```

Comment:
Note to reviewers - returning true signals failure in these functions.

Reply:
We should state what kind of field we are expecting, be it integers or strings.

Reply:
I don't follow, we do state in the error string that we expect a list.

Reply:
If the correct field is a list of strings, when provided a list of integers we error saying "expected list field", which is the wrong message. We should state we expected a list of strings ( even better also state what we got )

Reply:
I thought I left my original comment on line 70, where we do the check on the `T` type of the list, which is what should have the better diagnostic.

Reply:
Centralized the field type checks, now all of them should print detailed info about the failing check.